### ISSUE TITLE
Some questions about calculating PCs

### ISSUE THREAD

**Issue by HelloWorldLTY** [shbrief/GenomicSuperSignature#23]
![image](https://user-images.githubusercontent.com/43333475/179451263-4202fa7e-96d1-4524-b168-15dc772eda91.png)
Hi, I look the PCA.R code in the model building part again and I wonder if you used the loadings matrix as PCs to construct RAV models.

I also wonder why you did not run prcomp(x) directly so that you can directly get PCs as gene x PCs matrix. (like in python, we used PCA.fit_transform(x))? Thanks a lot!

**Comment by shbrief** [comment]
> I wonder if you used the loadings matrix as PCs to construct RAV models.   

Yes. Top 8 PCs from the loadings matrix (`rotation` value of `prcomp` output) were used for RAVmodel building.

> I also wonder why you did not run prcomp(x) directly so that you can directly get PCs as gene x PCs matrix. (like in python, > we used PCA.fit_transform(x))? 

I'm not sure what you mean by 'directly' here.

**Comment by HelloWorldLTY** [comment]
Thanks for your explanation. I mean I am confused about using t(x) rather than x. If we use prcomp(x), we can get gene x PCs matrix, I think it is also our target. Did you intend to centralize gene data so you used a loading matrix? Thanks a lot.

**Comment by shbrief** [comment]
I transposed the input matrix (`t(x)`) to make the variables (genes in this case) in the column, so the loading of `prcomp()` is genes x PCs matrix. FYI, genomic data often have more variables than observation, so variables are in the row which is different from the convention (observations x variables matrix) that `prcomp()` is assuming.

`prcomp()` was applied with the default parameters - centered but not scaled. You can find more details on RAVmodel building process in our paper: https://www.nature.com/articles/s41467-022-31411-3.
